harness/determined/experimental/client.py

```diff
@@ -233,7 +233,7 @@ def get_model_by_id(model_id: int) -> Model:
         model_id (int): The unique id of the model.
```

Comment:
can you also deprecate `client.get_model_by_id()`, separately from `Determined.get_model_by_id()`?

What I mean is:
- mark it as deprecated in this doc string
- mark issue a distinct deprecation warning
- make client.get_model_by_id() call `_determined.get_model()` directly, so that a user who calls `client.get_model_by_id()` isn't hit with two deprecation warnings for a single call.

Basically, my reasoning is that when a user calls a public API function and they get a deprecation warning for some other function called by that public API function, it looks more like a bug in the library and less like a useful message.

Reply:
✅  This reminded me to allow `client.get_model()` to accept both types as well